Dockerfile

```diff
@@ -54,11 +54,39 @@ RUN mkdir -p /artifacts && \
     /root/.cargo/bin/cargo build --profile=release && \
     cp target/release/libcorerust.so /artifacts/libcorerust.so
 
-FROM radixdlt/babylon-node-build-layers:${IMAGE_VERSION}-app AS main
+# =================================================================================================
+# LAYER: library-container
+# A layer containing just the built library at the root: /libcorerust.so
+# =================================================================================================
+FROM scratch AS library-container
+COPY --from=library-build-stage /artifacts/libcorerust.so /
+
+# =================================================================================================
+# LAYER: java-container
+# Exports only the java application artifacts from the java-build-stage
+# =================================================================================================
+FROM scratch AS java-container
+COPY --from=java-build-stage /radixdlt/core/build/distributions /
+
+# =================================================================================================
+# LAYER: library-builder-local
+# This layer is just used for local building in development via `local-cached-rust-build.yaml`.
+# Specifically - the Rust isn't built as part of the image, instead the CMD of the image is to do the build.
+# It allows us to use volumes at runtime to cache the build dependencies and artifacts.
+# =================================================================================================
+FROM radixdlt/babylon-node-build-layers:${IMAGE_VERSION}-rust AS library-builder-local
```

Comment:
Ah dang this is for the buildRustForDocker which is now only used by the buildDeb stuff.

Ideally we'd rename that to `buildRustForDeb` or something, but doesn't need to be this PR. And actually I think it makes more sense to nuke all this and potentially just run a standard cargo build on the target Debian-compatible system. Anyway, as I say, doesn't need to be this PR :)